lib/engine/game/g_1893/step/buy_train.rb

```diff
@@ -1,12 +1,31 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/buy_train'
+require_relative '../buy_eva_from_other_player'
 
 module Engine
   module Game
     module G1893
       module Step
         class BuyTrain < Engine::Step::BuyTrain
+          def actions(entity)
+            possible_actions = super
+
+            return [] if possible_actions.empty?
+
+            possible_actions << 'buy_company'
+          end
+
+          include BuyEvaFromOtherPlayer
+
+          # Buy EVA from other player
+          def process_buy_company(action)
+            company = action.company
+            raise GameError, "Only #{@game.eva.name} can be bought from other player" unless company == @game.eva
```

Comment:
why do you do this check twice

Reply:
Right, not needed there.
Having the check in buy_eva_from_other_player might still be good as that method can be called from stock round code that might have other companies you can buy.

I put some guards in actions in buy_train.rb so that action is only available if a purchase is possible.